### PR TITLE
Setup server

### LINKED ISSUE #1: set up server
build the server using express framework .

- make the errors middle-wars

### PR DESCRIPTION
**Pull request by AlaaShurrab**
created the basic listen and lunched the server

### CODE REVIEW

**Review by imansedky5 — APPROVED** [review]

File: src/app.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const { e404, serverErr } = require('./controllers/err');
+
+const app = express();
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+
+app.use(express.static(join(__dirname, '..', 'public')));
+app.use(cookiePerser);
+// app.use(rauter);
```
Comment (yasmeenattallah) [inline src/app.js:16]: remove comment line

**Review by mAbadsa — APPROVED** [review]

**Review by yasmeenattallah — APPROVED** [review]

File: src/app.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const express = require('express');
+const { join } = require('path');
+const cookiePerser = require('cookie-parser')
+const rauter = require('./router');
+const { e404, serverErr } = require('./controllers/err');
```
Comment (AhmedSafi97) [inline src/app.js]: relate to this issue please [#29](https://github.com/GSG-G9/IM-weather-app/issues/29)

File: src/app.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+
+app.use(express.static(join(__dirname, '..', 'public')));
+app.use(cookiePerser);
+// app.use(rauter);
+
+app.use(e404);
+app.use(serverErr);
```
Comment (AhmedSafi97) [inline src/app.js]: I would call it errHandler rather than server error since it will catch different type of errors

File: src/app.js (hunk 1 above)
Comment (AhmedSafi97) [reply src/app.js:16]: actually you should set up the router file

**Review by AhmedSafi97 — APPROVED** [review]

File: src/app.js (hunk 1 above)
Comment (AlaaShurrab) [reply src/app.js:16]: ill do it but i need to work with Muhammed with the database queries to make it right

File: src/app.js (hunk 2 above)
Comment (AlaaShurrab) [reply src/app.js]: is this better?

File: src/app.js (hunk 3 above)
Comment (AlaaShurrab) [reply src/app.js]: done

File: src/app.js (hunk 1 above)
Comment (AlaaShurrab) [reply src/app.js:16]: I want to work on the router when Muhammed is done with the database queries

**Review by mAbadsa — APPROVED** [review]

**Review by mAbadsa — APPROVED** [review]

**Pull request merged**